pkg/apis/pipeline/v1beta1/result_defaults.go

```diff
@@ -21,4 +21,14 @@ func (tr *TaskResult) SetDefaults(context.Context) {
 		// ResultsTypeString is the default value
 		tr.Type = ResultsTypeString
 	}
+	// Set type to object if `properties` is given
+	if tr.Properties != nil {
+		tr.Type = ResultsTypeObject
+	}
+	// Set default type of object values to string
+	for key, propertySpec := range tr.Properties {
+		if propertySpec.Type == "" {
+			tr.Properties[key] = PropertySpec{Type: ParamType(ResultsTypeString)}
+		}
+	}
 }
```

Comment:
If the result type is specified, I assume this `SetDefaults` will set whatever type users defined right?

If so, what if the type specified is `string` or `array`, but `properties` section is also provided. What type would this result be?

Reply:
ah yes, good point